mpas_analysis/shared/mpas_xarray/mpas_xarray.py

```diff
@@ -141,59 +141,114 @@ def ensure_list(alist):  # {{{
     return alist  # }}}
 
 
-def get_datetimes(ds, timestr, yearoffset):  # {{{
+def parse_time(ds, inTimeVarName, inReferenceDate, outTimeVarName,
+               outReferenceDate):  # {{{
     """
-    Computes a list of datetimes from the time variable in the dataset ds with
-    variable name (or a list of 2 names) given by timestr, typically one of
-    'daysSinceStartOfSim', 'xtime', or ['xtime_start', 'xtime_end'].
+    Converts MPAS times to `netcdftime.DatetimeNoLeap` objects consistent with
+    MPAS's 365-day calendar.
```

Comment:
My understanding is that MPAS supports multiple calendar types, e.g., 'gregorian' and 'gregorian_noleap'.  We need to support both options.  The attribute is called `config calendar type` on the ocean side.  Regardless, we are really talking about enhancing generalization for this PR so we need to have more general support for calendar types.